### PR TITLE
Neat 257 create validation module for information rules

### PR DESCRIPTION
**Pull request by nikokaoja**
Information Rules are now post-model validated using dedicated module.

A lot of small fixes due to more restrictive validations.

### CODE REVIEW

File: tests/tests_legacy/tests_rules/importer/test_ontology2rules.py [hunk 1]
```diff
@@ -1,11 +1,11 @@
+from cognite.neat.legacy.rules.importers._owl2rules import OWLImporter
 from cognite.neat.rules.examples import wind_energy_ontology
-from cognite.neat.rules.importers._owl2rules import OWLImporter
```
Comment (nikokaoja) [inline tests/tests_legacy/tests_rules/importer/test_ontology2rules.py:4]: Legacy test was tested with the current methods :)

File: cognite/neat/rules/importers/_dtdl2rules/spec.py [hunk 2]
```diff
@@ -316,7 +316,7 @@ def list_to_string(cls, value: Any) -> Any:
     def select_content_type(cls, value: Any, info) -> Any:
         if not isinstance(value, list):
             return value
-        context = info.data.get("@context", cls.default_context)
+        context = info.data.get("context", cls.default_context)
```
[the remark this answers is not in the document]
Comment (doctrino) [reply cognite/neat/rules/importers/_dtdl2rules/spec.py:319]: Reverting this and rerunning the test passes.

**Review by doctrino — COMMENTED** [review]
I think two of the changes should likely be reverted the `@context` and `INFORMATION_UNKNOWN_VALUE_TYPE` tests.

File: cognite/neat/rules/importers/_dtdl2rules/spec.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (nikokaoja) [reply cognite/neat/rules/importers/_dtdl2rules/spec.py]: Returned

**Pull request merged**